ProjFS.Mac/PrjFSKext/PrjFSKext/KextLog.hpp

```diff
@@ -42,6 +44,53 @@ template <typename... args>
 #define KextLog_FileInfo(vnode, format, ...)  ({ _os_log_verify_format_str(format, ##__VA_ARGS__); KextLogFile_Printf(KEXTLOG_INFO, vnode, format " (vnode path: '%s')", ##__VA_ARGS__); })
 #define KextLog_FileNote(vnode, format, ...)  ({ _os_log_verify_format_str(format, ##__VA_ARGS__); KextLogFile_Printf(KEXTLOG_NOTE, vnode, format " (vnode path: '%s')", ##__VA_ARGS__); })
 
+template <typename... args>
+    void KextLog_PrintfVnodePathAndProperties(KextLog_Level loglevel, struct vnode* vnode, const char* fmt, args... a)
+    {
+        char vnodePath[PrjFSMaxPath] = "";
+        int vnodePathLength = PrjFSMaxPath;
+        vn_getpath(vnode, vnodePath, &vnodePathLength);
+        
+        const char* name = vnode_getname(vnode);
+        mount_t mount = vnode_mount(vnode);
+        vfsstatfs* vfsStat = mount != nullptr ? vfs_statfs(mount) : nullptr;
+        
+        KextLog_Printf(loglevel, fmt, a..., vnodePath, name ?: "[NULL]", vnode_vtype(vnode), vnode_isrecycled(vnode) ? "yes" : "no", vfsStat ? vfsStat->f_mntonname : "[NULL]");
+ 
+        if (name != nullptr)
+        {
+            vnode_putname(name);
```

Comment:
I like the way you've done this change :-).  Could you please add a small comment above the vnode_putname explaining why we need it.  I imagine it's redundant for folks that know kext well but for newbies like me it'd be helpful.

Reply:
Done!